### PR TITLE
[Evo] #5287 processing statistics screen added + statistics calculed …

### PR DESCRIPTION
**Pull request by laurianemas**
…from registered events on general rabbitmq listener

WARN: DC Importer processing statistics are registered by a general rabbitmq listener from the day you install this version. To see something you must do process or manually fullfill ProcessingStat document in your dev mongo database

MISSING : 
- Unit tests
- Rebase on Develop
- Visual improvements on statistic screen (maybe bars graph ?)

### CODE REVIEW

File: src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    fun create(processingStat: ProcessingStat){
+        processingStatRepository.save(processingStat).subscribe()
+    }
+
+    fun getAll(): Flux<ProcessingStat> {
+        return processingStatRepository.findAll()
+    }
+
+    //No use yet but just in case, it's here
```
Comment (bobeal) [inline src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt]: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

File: src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                processingStatRepository.findByOrganizationAndAction(organization, action)
+            //model+organization+action+date
+            !date.isEmpty() && !model.isEmpty() && !organization.isEmpty() && !action.isEmpty() ->
+                processingStatRepository.findByModelAndOrganizationAndActionAndCreationDateAfter(model, organization, action, stringToLocalDateTime(date))
+            else -> processingStatRepository.findAll()
+        }
+    }
+
+    fun getByDate(date: LocalDateTime): Flux<ProcessingStat> {
+        return processingStatRepository.findByCreationDateAfter(date)
```
Comment (bobeal) [inline src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt:24]: the "date after" thing is not reflected in the method name. that can cause some usability problems.

File: src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    }
+
+    fun getGeneralStat(date: String): Mono<ProcessingResumeFields>{
```
Comment (bobeal) [inline src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt:27]: you'are doing a lot of things in this method but that's ok for this very first iteration

File: src/main/kotlin/org/ozwillo/dcimporter/config/RabbitMQConfig.kt [hunk 4]
```diff
@@ -82,4 +82,21 @@ class RabbitMQConfig {
             return BindingBuilder.bind(queue).to(topic).with(BINDING_KEY)
         }
     }
+
+    private class ProcessingStatReceiverConfig {
```
Comment (nabdali) [inline src/main/kotlin/org/ozwillo/dcimporter/config/RabbitMQConfig.kt:86]: In my opinion it's overkill to create a new queue just for stats, we can do that in concerned sender

File: src/main/kotlin/org/ozwillo/dcimporter/repository/ProcessingStatRepository.kt [hunk 5]
```diff
@@ -0,0 +1,88 @@
+package org.ozwillo.dcimporter.repository
+
+import org.ozwillo.dcimporter.model.ProcessingStat
+import org.springframework.data.mongodb.repository.ReactiveMongoRepository
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import java.time.LocalDateTime
+
+@Repository
+interface ProcessingStatRepository: ReactiveMongoRepository<ProcessingStat, String>{
```
Comment (nabdali) [inline src/main/kotlin/org/ozwillo/dcimporter/repository/ProcessingStatRepository.kt:10]: We can do dynamic research, using criterias or queryDSL 
Helpfull link : https://www.baeldung.com/queries-in-spring-data-mongodb

File: src/main/resources/public/src/views/ProcessingStat.vue [hunk 6]
```diff
@@ -0,0 +1,374 @@
+<template>
+    <div class="container">
+        <h2>Processing Statistics</h2>
+        <div class="form-group row">
```
Comment (nabdali) [inline src/main/resources/public/src/views/ProcessingStat.vue:4]: it's not a form

File: src/main/resources/public/src/views/ProcessingStat.vue [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+            <label for="claimer-stat-display-date" class="col-sm-3 col-form-label col-form-label-sm">
+                From: 
+            </label>
+            <input type="date" id="claimer-stat-display-date" v-model="datePicker"/>
+        </div>
+        <div class="form-group row">
+            <div class="general-stat">
+                <p class="p-text">DC Importer has completed
```
Comment (nabdali) [inline src/main/resources/public/src/views/ProcessingStat.vue]: It's badly indented, and `<br/>` is overused ..

File: src/main/kotlin/org/ozwillo/dcimporter/config/RabbitMQConfig.kt (hunk 4 above)
Comment (bobeal) [reply src/main/kotlin/org/ozwillo/dcimporter/config/RabbitMQConfig.kt:86]: it's preferable to decouple the stats processing (SoC)

File: src/main/resources/public/src/views/ProcessingStat.vue (hunk 6 above)
Comment (laurianemas) [reply src/main/resources/public/src/views/ProcessingStat.vue:4]: Yep but I wanted to keep the exact form style and organization. Even if "form group" don't have a real impact on date type input visual it provides the good spacing between the different div tag. Is it a problem to use it ?

File: src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt (hunk 1 above)
Comment (laurianemas) [reply src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt]: Removed

File: src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt (hunk 2 above)
Comment (laurianemas) [reply src/main/kotlin/org/ozwillo/dcimporter/service/ProcessingStatService.kt:24]: Changed to getByDateAfter(date)

File: src/main/kotlin/org/ozwillo/dcimporter/repository/ProcessingStatRepository.kt (hunk 5 above)
Comment (laurianemas) [reply src/main/kotlin/org/ozwillo/dcimporter/repository/ProcessingStatRepository.kt:10]: As the "search" function have no utility yet I removed it. But I'll keep this in mind, thank you.

File: src/main/resources/public/src/views/ProcessingStat.vue (hunk 7 above)
Comment (laurianemas) [reply src/main/resources/public/src/views/ProcessingStat.vue]: I tried to clean it a little tell me if more is needed

**Pull request merged**